src/org/thoughtcrime/securesms/ConversationFragment.java

```diff
@@ -326,17 +326,24 @@ public int compare(MessageRecord lhs, MessageRecord rhs) {
   }
 
   public void handlePinOrUnpinMessage(final MessageRecord message, boolean pin,
-                                      PinnedMessagesHandler handler) {
-    PinnedMessagesHandler pinHandler;
+                                      PinnedMessageHandler handler) {
+    PinnedMessageHandler  pinHandler;
     MessagingDatabase     databaseToQuery;
     String                outputMessage;
     boolean               result;
 
     pinHandler       = handler;
     databaseToQuery  = pinHandler.getAppropriateDatabase(message);
 
-    if(!message.isMms()) {
-      if (pin) {
+    if (pin) {
+      // Blocking Video and Audio pinning Temporarily
+        if(message.isMms()) {
+          MediaMmsMessageRecord mediaMessage = (MediaMmsMessageRecord) message;
+          if (!mediaMessage.getSlideDeck().getThumbnailSlide().getContentType().contains("image")) {
```

Comment:
Is GIF considered image? Does it pin successfully? 
As well, does documents crash it too?

Reply:
@shelbaz yes, it pinning Gif works, But the document is still breaking